Adapter/Algolia.php

```diff
@@ -78,7 +93,10 @@ public function __construct(
         ConfigHelper $config,
         Data $catalogSearchHelper,
         StoreManagerInterface $storeManager,
+        Registry $registry,
+        CustomerSession $customerSession,
         AlgoliaHelper $algoliaHelper,
+        ProductHelper $productHelper,
         Http $request,
         DocumentFactory $documentFactory
```

Comment:
Great job, on the class, @damcou! It looks really good.

One thing - could you try to refactor / split the class to multiple smaller classes as the class has 9 dependencies,
now. This is really not great for a future maintain ability and we should keep the number of class (constructor) dependencies as low as possible. A single class should have [max 4-5 dependencies](https://stackoverflow.com/questions/953407/constructor-injection-how-many-dependencies-is-too-many) as it usually means it breaks [Single responsibility principle](https://en.wikipedia.org/wiki/Single_responsibility_principle).
A good way to start splitting is to identify methods which uses a common dependencies and move them to a more self-descriptive smaller class and inject only the newly created class.

I know, the extension is full "large" constructors, but for the future it makes sense to start refactoring it now and we need to start somewhere :) Thank you!